internal/cephes/lanczos.go

```diff
@@ -0,0 +1,180 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+ * Cephes Math Library Release 2.1:  December, 1988
+ * Copyright 1984, 1987, 1988 by Stephen L. Moshier
+ * Direct inquiries to 30 Frost Street, Cambridge, MA 02140
+ */
+
+/*
+ * Adapted from scipy's cephes lanczos.c
+ */
+
+/* Both lanczos.h and lanczos.c were formed from Boost's lanczos.hpp
+ *
+ * Scipy changes:
+ * - 06-22-2016: Removed all code not related to double precision and
+ *   ported to c for use in Cephes. Note that the order of the
+ *   coefficients is reversed to match the behavior of polevl.
+ */
+
+/*
+ * Optimal values for G for each N are taken from
+ * http://web.mala.bc.ca/pughg/phdThesis/phdThesis.pdf,
+ * as are the theoretical error bounds.
+ *
+ * Constants calculated using the method described by Godfrey
+ * http://my.fit.edu/~gabdo/gamma.txt and elaborated by Toth at
+ * http://www.rskey.org/gamma.htm using NTL::RR at 1000 bit precision.
+ */
+
+/*
+ * Lanczos Coefficients for N=13 G=6.024680040776729583740234375
+ * Max experimental error (with arbitary precision arithmetic) 1.196214e-17
+ * Generated with compiler: Microsoft Visual C++ version 8.0 on Win32 at Mar 23 2006
+ *
+ * Use for double precision.
+ */
+
+package cephes
+
+var lanczosNum = []float64{
+	2.506628274631000270164908177133837338626,
+	210.8242777515793458725097339207133627117,
+	8071.672002365816210638002902272250613822,
+	186056.2653952234950402949897160456992822,
+	2876370.628935372441225409051620849613599,
+	31426415.58540019438061423162831820536287,
+	248874557.8620541565114603864132294232163,
+	1439720407.311721673663223072794912393972,
+	6039542586.35202800506429164430729792107,
+	17921034426.03720969991975575445893111267,
+	35711959237.35566804944018545154716670596,
+	42919803642.64909876895789904700198885093,
+	23531376880.41075968857200767445163675473,
+}
+
+var lanczosDenom = []float64{
+	1,
+	66,
+	1925,
+	32670,
+	357423,
+	2637558,
+	13339535,
+	45995730,
+	105258076,
+	150917976,
+	120543840,
+	39916800,
+	0,
+}
+
+var lanczosSumExpgScaledNum = []float64{
```

Comment:
I had them as arrays originally but `ratevl` in `polevl.go` expects slices. So I could define them as arrays and then turn them into slices when passing them to `ratevl`. Is that preferable?